In [ ]:
import pandas as pd
from IPython.display import display, HTML

pd.set_option('display.max_columns', None)
pd.set_option('display.expand_frame_repr', False)
pd.set_option('display.max_colwidth', None)

variables_df = pd.read_excel("../external/PAKDD2010_VariablesList.XLS")
column_names = variables_df['Var_Title'].head(53).tolist()

data_df = pd.read_csv("../external/PAKDD2010_Leaderboard_Data.txt", sep='\t', header=None)

# mergeamos cabeceras con datos
data_df.columns = column_names

# mostrar primeras 10
display(HTML(data_df.head(10).to_html(index=False)))


ImportError: Missing optional dependency 'xlrd'. Install xlrd >= 2.0.1 for xls Excel support Use pip or conda to install xlrd.

Anánlisis de datos



In [9]:

print(data_df.shape)
print(data_df.dtypes)



(20000, 53)
ID_CLIENT                           int64
CLERK_TYPE                         object
PAYMENT_DAY                         int64
APPLICATION_SUBMISSION_TYPE        object
QUANT_ADDITIONAL_CARDS              int64
POSTAL_ADDRESS_TYPE                 int64
SEX                                object
MARITAL_STATUS                      int64
QUANT_DEPENDANTS                    int64
EDUCATION_LEVEL                     int64
STATE_OF_BIRTH                     object
CITY_OF_BIRTH                      object
NACIONALITY                        object
RESIDENCIAL_STATE                  object
RESIDENCIAL_CITY                   object
RESIDENCIAL_BOROUGH                object
FLAG_RESIDENCIAL_PHONE             object
RESIDENCIAL_PHONE_AREA_CODE        object
RESIDENCE_TYPE                    float64
MONTHS_IN_RESIDENCE               float64
FLAG_MOBILE_PHONE                  object
FLAG_EMAIL                          int64
PERSONAL_MONTHLY_INCOME           float64
OTHER_INCOMES         

Análisis de nulos


In [8]:
print(data_df.isnull().sum())

ID_CLIENT                             0
CLERK_TYPE                            0
PAYMENT_DAY                           0
APPLICATION_SUBMISSION_TYPE           0
QUANT_ADDITIONAL_CARDS                0
POSTAL_ADDRESS_TYPE                   0
SEX                                   0
MARITAL_STATUS                        0
QUANT_DEPENDANTS                      0
EDUCATION_LEVEL                       0
STATE_OF_BIRTH                        0
CITY_OF_BIRTH                        10
NACIONALITY                           0
RESIDENCIAL_STATE                     1
RESIDENCIAL_CITY                      1
RESIDENCIAL_BOROUGH                   1
FLAG_RESIDENCIAL_PHONE                0
RESIDENCIAL_PHONE_AREA_CODE           0
RESIDENCE_TYPE                      616
MONTHS_IN_RESIDENCE                2663
FLAG_MOBILE_PHONE                     0
FLAG_EMAIL                            0
PERSONAL_MONTHLY_INCOME               0
OTHER_INCOMES                         0
FLAG_VISA                             0



Podríamos descartar estos 4 por la cantidad de nulos:

MATE_PROFESSION_CODE              18729
EDUCATION_LEVEL                   19539
PROFESSIONAL_CITY                 14206
PROFESSIONAL_BOROUGH              14206

Tenemos que establecer una estrategia para los que tienen pocos nulos:

CITY_OF_BIRTH                        10
RESIDENCIAL_STATE                     1
RESIDENCIAL_CITY                      1
RESIDENCIAL_BOROUGH                   1
RESIDENCE_TYPE                      616
MONTHS_IN_RESIDENCE                2663
OCCUPATION_TYPE                    2437


mediana para las numéricas  ?
más frecuente para las categóricas ?

Revisar variables categóricas:





In [ ]:
print(df['columna_categ'].value_counts())


limpiar error de tipeo

In [ ]:
df['columna'] = df['columna'].str.lower().str.strip()


Codificación de variables categóricas para modelos lineares

In [ ]:
from sklearn.preprocessing import OneHotEncoder
df = pd.get_dummies(df, columns=['columna_categ'])


para árboles

In [ ]:
df['columna_categ'] = df['columna_categ'].astype('category').cat.codes


DETECTAR OUTLIERS

REEMPLAZAR POR PERCENTILES ?

NORMALIZACÓN Y ESCALADO.